devicetypes/vision-nameseca/zp3111jp-4-in-1-motion.src/zp3111jp-4-in-1-motion.groovy

```diff
@@ -0,0 +1,925 @@
+			[min: 50, max: 59.99, multiplier: 0.6474],
+			[min: 60, max: 69.99, multiplier: 0.5222],
+			[min: 70, max: 79.99, multiplier: 0.5204],
+			[min: 80, max: 89.99, multiplier: 0.4965],
+			[min: 90, max: 100, multiplier: 0.4843]
+	]
+}
+private applyOffset(val, offsetVal, name, unit) {
+	if (offsetVal) {
+		logDebug "Before Applying ${offsetVal}${unit} ${name} Offset to ${val}${unit}"
+		val = (safeToDec(val, 0) + safeToDec(offsetVal, 0))
+		logDebug "After Applying ${offsetVal}${unit} ${name} Offset to ${val}${unit}"
+	}
+	return roundVal(val, decimalPlacesSetting)
+}
+private createStatusEventMaps(eventMaps, onlyIfNew) {
+	def result = []
+
+	def primaryStatus = eventMaps?.find { it.name == primaryTileStatusSetting }?.descriptionText
+	if (primaryStatus) {
+		if (roundPrimaryStatusSetting) {
+			primaryStatus = formatPrimaryStatusNumber(primaryStatus)
+		}
+		result += createEventMaps("primaryStatus", primaryStatus, "", false, onlyIfNew)
+	}
+
+	def secondaryStatus = getSecondaryStatus(eventMaps)
+	if (secondaryStatus || secondaryTileStatusSetting == "none") {
+		result += createEventMaps("secondaryStatus", secondaryStatus, "", false, onlyIfNew)
+	}
+	return result
+}
+private formatPrimaryStatusNumber(val) {
+	def unit
+	["% LIGHT", "% RH", " LUX", "°F", "°C"].each {
+		if ("${val}".contains(it)) {
+			unit = "${it}"
+		}
+	}
+
+	if (unit) {
+		def numericVal = safeToDec("${val}".replace("${unit}", ""))
+		return "${Math.round(numericVal)}${unit}"
+	}
+	else {
+		return val
+	}
+}
+
+def clearTamper() {
+	logDebug "Clearing Tamper"
+	sendEvent(createTamperEventMap("clear"))
+	return []
+}
+
+private getSecondaryStatus(eventMaps) {
+	def status = ""
+	if (secondaryTileStatusSetting == "combined"){
+		def motionStatus = getAttrStatusText("motion", eventMaps)
+		def lightStatus = getAttrStatusText("lxLight", eventMaps)
+		def tempStatus = getAttrStatusText("temperature", eventMaps)
+		def humidityStatus = getAttrStatusText("humidity", eventMaps)
+		status = "${motionStatus} / ${tempStatus} / ${humidityStatus} / ${lightStatus}"
+	}
+	else if (status != "none") {
+		status = getAttrStatusText(secondaryTileStatusSetting, eventMaps)
+	}
+	return status
+}
+private getAttrStatusText(attrName, eventMaps=null) {
+	def status = (eventMaps?.find { it.name == attrName }?.descriptionText)
+	if (status) {
+		return status
+	}
+	else {
+		return getDescriptionText(device.currentState(attrName))
+	}
+}
+private getDescriptionText(data) {
+	switch (data?.name ?: "") {
+		case "motion":
+			return "${data.value}" == "active" ? "Motion" : "No Motion"
+			break
+		case "temperature":
+			return "${data.value}°${data.unit}"
+			break
+		case "humidity":
+			return  "${data.value}% RH"
+			break
+		case "lxLight":
+			return "${data.value} LUX"
+		case "pLight":
+			return "${data.value}% LIGHT"
+			break
+		default:
+			return ""
+	}
+}
+
+def refresh() {
+	log.debug "sending battery refresh command"
+	batteryGetCmd()
```

Comment:
You can potentially set a flag here that a refresh is required (or maybe set `state.refreshAll` to true) and when the device wakes up request the latest and greatest?

Reply:
I already useed state.refreshAll to true on  these programs.
thank you.